src/pages/[eventAbbr]/ui/admin/check_in_event/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { NextPage } from 'next'
+import React, { useMemo } from 'react'
+import { Layout } from '../../../../../components/Layout'
+import { useGetApiV1EventsByEventAbbrQuery } from '../../../../../generated/dreamkast-api.generated'
+import { useRouter } from 'next/router'
+import { useSelector } from 'react-redux'
+import { authSelector } from '../../../../../store/auth'
+import { withAuthProvider } from '../../../../../context/auth'
+import Error404 from '../../../../404'
+import { CheckIn } from '../../../../../components/CheckIn/CheckIn'
+import { Typography } from '@material-ui/core'
+
+const IndexPage: NextPage = () => {
+  return withAuthProvider(<IndexMain />)
+}
+
+const IndexMain = () => {
+  const router = useRouter()
+  const eventAbbr = useMemo<string>(() => {
+    if (router.asPath !== router.route) {
+      const { eventAbbr } = router.query
+      return eventAbbr as string
+    }
+    return ''
+  }, [router])
+  const { roles } = useSelector(authSelector)
+  const skip = eventAbbr === null
+  const { data: event } = useGetApiV1EventsByEventAbbrQuery(
+    { eventAbbr },
+    { skip },
+  )
+
+  const includeAdminRole = roles.some((role) =>
+    [`${eventAbbr.toUpperCase()}-Admin`].includes(role),
+  )
```

Comment:
```suggestion
  const includeAdminRole = roles.includes(`${eventAbbr.toUpperCase()}-Admin`) 
```

これでいい気がしました。
あと、eventAbbrがnullのケースを考慮できていない気がするので、先にガード節で弾いても良さそうです。

Reply:
あれ、今見るとなんか変なコードを書いてましたね…suggested changeでよさそうですね。

> あと、eventAbbrがnullのケースを考慮できていない気がするので、先にガード節で弾いても良さそうです。

eventAbbrの型はstringだったで大丈夫そうですかね？

![image](https://github.com/cloudnativedaysjp/dreamkast-ui/assets/107187/07c5c813-6f0b-4585-8724-2977be757628)